apps/encryption/lib/Crypto/Encryption.php

```diff
@@ -462,15 +471,29 @@ public function shouldEncrypt($path) {
 	 * get size of the unencrypted payload per block.
 	 * Nextcloud read/write files with a block size of 8192 byte
 	 *
+	 * Encrypted blocks have a 22-byte IV and 2 bytes of padding, encrypted and
+	 * signed blocks have also a 71-byte signature and 1 more byte of padding,
+	 * resulting respectively in:
+	 *
+	 *  8192 - 22 - 2 = 8168 bytes in each unsigned unencrypted block
+	 *  8192 - 22 - 2 - 71 - 1 = 8096 bytes in each signed unencrypted block
+	 *
+	 * Legacy base64 encoding then reduces the available size by a 3/4 factor:
+	 *
+	 *  8168 * (3/4) = 6126 bytes in each base64-encoded unsigned unencrypted block
+	 *  8096 * (3/4) = 6072 bytes in each base64-encoded signed unencrypted block
+	 *
 	 * @param bool $signed
 	 * @return int
 	 */
 	public function getUnencryptedBlockSize($signed = false) {
-		if ($signed === false) {
-			return $this->unencryptedBlockSize;
+		$unencryptedBlockSize = $signed ? 8096 : 8168;
+
+		if ($this->useLegacyBase64Encoding) {
+			$unencryptedBlockSize = $unencryptedBlockSize * 3 / 4;
 		}
 
-		return $this->unencryptedBlockSizeSigned;
+		return $unencryptedBlockSize;
```

Comment:
This line make it seems that the `unencryptedBlockSize` is dynamic where it is static. Would it make sense to be more explicit?

```suggestion
		if ($this->useLegacyBase64Encoding) {
			if ($signed) {
				return 6072;
			} else {
				return 6126;
			}
		} else {
			if ($signed) {
				return 8096;
			} else {
				return 8168;
			}
		}

```

Reply:
Not even necessary to store the value in a tmp variable

Reply:
Just pushed a version.